internal/versionsapi/version.go

```diff
@@ -142,9 +144,32 @@ func (v Version) ListPath(gran Granularity) string {
 	)
 }
 
-// ArtifactURL returns the URL to the artifacts stored for this version.
+// MeasurementURL builds the measurement and signature URLs for the given CSP.
+func (v Version) MeasurementURL(csp cloudprovider.Provider) (*url.URL, *url.URL, error) {
+	measurementPath, err := url.JoinPath(v.ArtifactsURL(), "image", "csp", strings.ToLower(csp.String()), constants.CDNMeasurementsFile)
+	if err != nil {
+		return &url.URL{}, &url.URL{}, fmt.Errorf("joining path for measurement: %w", err)
+	}
+	signaturePath, err := url.JoinPath(v.ArtifactsURL(), "image", "csp", strings.ToLower(csp.String()), constants.CDNMeasurementsSignature)
+	if err != nil {
+		return &url.URL{}, &url.URL{}, fmt.Errorf("joining path for signature: %w", err)
+	}
+
+	measurementURL, err := url.Parse(measurementPath)
+	if err != nil {
+		return &url.URL{}, &url.URL{}, fmt.Errorf("parsing path for measurement: %w", err)
+	}
+
+	signatureURL, err := url.Parse(signaturePath)
+	if err != nil {
+		return &url.URL{}, &url.URL{}, fmt.Errorf("parsing path for signature: %w", err)
+	}
+	return measurementURL, signatureURL, nil
+}
```

Comment:
This can't be implemented as method of the version struct. The measurement artifact is special to a image version, but the version struct represents versions of multiple kind. There is no such thing as a measuremntURL for a version of versionKindCLI.

Reply:
Made this a function that receives a function as argument. Error out for versions that are of wrong kind.